packages/ckeditor5-ckbox/src/ckboxcommand.ts

```diff
@@ -449,16 +457,36 @@ function isImage( asset: CKBoxRawAssetDefinition ) {
 /**
  * Creates the URL for the asset.
  *
- * @param origin The base URL for assets inserted into the editor.
+ * @param config The CKBox download asset configuration.
+ * @param asset The asset to create the URL for.
  */
-function getAssetUrl( asset: CKBoxRawAssetDefinition ) {
+function getAssetUrl( config: CKBoxConfig[ 'downloadableFiles' ], asset: CKBoxRawAssetDefinition ) {
 	const url = new URL( asset.data.url );
 
-	url.searchParams.set( 'download', 'true' );
+	if ( isDownloadableAsset( config, asset ) ) {
+		url.searchParams.set( 'download', 'true' );
+	}
 
 	return url.toString();
 }
 
+/**
+ * Determines if download should be enabled for given asset based on configuration.
+ *
+ * @param config The CKBox download asset configuration.
+ * @param asset The asset to check.
+ */
+function isDownloadableAsset(
```

Comment:
The same comment as for `prepareLinkAssetAttributes` function, I'd change the order of parameters to `( asset, config )`.